src/softmax.jl

```diff
@@ -142,3 +142,9 @@ function logsumexp(x::AbstractArray; dims = :)
     max_ = maximum(x; dims = dims)
     max_ .+ log.(sum(exp.(x .- max_); dims = dims))
 end
+
+# Informative error message if any of the softmax variants is called with a number
+for f in (:softmax, :logsoftmax, :softmax!, :logsoftmax!, :logsumexp)
+    @eval $(f)(x::Number, args...) = 
+      error("(`", $(string(f)), "(x)`) called with a number, but it expects an array.")
```

Comment:
Should this be a warning instead? `softmax` is doing its job, but calling it with a number is not terribly useful. We added a documentation update saying that correct usage of `softmax` is on an array already too, specifically addressing the `Number` case.

Reply:
It won't work right now unless you call it like `softmax([x])` (see the linked issue).